sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/planning/TestPhysicalOperation.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.planning
+
+import org.junit.Assert
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.expressions.{Alias, AttributeReference, Expression, ExprId, IsNotNull}
+import org.apache.spark.sql.catalyst.plans.logical.{Filter, LocalRelation, LogicalPlan, Project}
+import org.apache.spark.sql.types.{LongType, StringType, TimestampType}
+
+class TestPhysicalOperation extends SparkFunSuite {
+  private val textField = AttributeReference("text1", StringType, nullable = false)()
+  private val dateField = AttributeReference("date1", TimestampType)()
+  private val longField = AttributeReference("long1", LongType)()
+  private val longAliasField = Alias(longField, "long1")(exprId = ExprId(100))
+  private val longAliasField2 =
+    Alias(longField, "long1")(exprId = ExprId(100), qualifier = Seq("tmp"))
+  private val testRelation = LocalRelation(textField, dateField, longField)
+
```

Comment:
Can we simplify the test? I think a very obvious consequence of this bug is: we can't push down certain filters to data source.

Reply:
Actually can we remove all the test cases? I don't think we guarantee anything internal to work with something external. I was wondering if we can consider this PR as a code cleanup.

Reply:
@cloud-fan could you explain more on how we can simplify the test, I'm a bit unclear. Also the other two test files `DataSourceStrategyWithCustomBarrierSuite.scala` and `CustomBarrier.scala` were added only to show the full reproducer of the bug, I can remove them if they are not needed.

Reply:
I think he meant a test case that causes the similar failure to show it is indeed a bug within Apache Spark without external stuff.

I think we should remove out such custom barriers - that's not what Apache Spark guarantees. Can we find a case where we can't push down certain filters to data source?

Reply:
ah I was wrong. Since optimizer always push filter through project, so we should never hit this bug.

Maybe just remove all tests and treat it as a code cleanup.

Reply:
Yes correct. Okay will remove all tests and update shortly.